internal/pkg/aws/session/session.go

```diff
@@ -30,11 +30,30 @@ func userAgentHandler() request.NamedHandler {
 	}
 }
 
-// Factory holds methods to create sessions.
-type Factory struct{}
+// Provider provides methods to create sessions.
+// Once a session is created, it's cached locally so that the same session is not re-created.
+type Provider struct {
+	defaultSess  *session.Session
+	regionCache  map[string]*session.Session
+	profileCache map[string]*session.Session
+	roleCache    map[string]*session.Session
```

Comment:
Do we envision there's a need for this many sessions? Otherwise this might be a bit overly complex.

Reply:
Discussed offline, only the `default` session is used multiple times across commands so we'll only cache that in the provider